src/engine/Globals.ml

```diff
@@ -36,16 +36,32 @@
 (* Entry point *)
 (*****************************************************************************)
 
+type pro_hook_ref = Pro_hook_ref : 'a option ref -> pro_hook_ref
```

Comment:
What is this type? GADT? Maybe a comment would be useful then why we need it

Reply:
Yep that's a GADT to create an existential type, so we can put in a list different hooks for which `'a` is different. I'll add that comment.